### DISCUSSION TITLE
Can you batch inpaint with this automatically?

### DISCUSSION THREAD

**Discussion by sahildeshpande** [continue-revolution/sd-webui-segment-anything#88] [General]
Or do you first have to make the masks using batch process and then batch inpaint separately?

**Comment by continue-revolution** [comment]
I think you can inpaint in batch at img2img/batch tab. You can try and see if it works. I've not tried this feature yet.

**Reply by sahildeshpande** [reply]
Thanks for replying. How do I combine it with Batch Process mode from this extension ?

**Reply by continue-revolution** [reply]
You firstly generate masks with sam extension, then you put in the input directory, output directory, inpaint batch mask directory inside img2img/batch tab.

**Reply by sahildeshpande** [reply]
Yeah sweet that’s what I thought. Thanks

**Reply by continue-revolution** [reply]
You probably have to rename in-batch for all the masks generated by sam extension to let them have the same filename as the input image. However I will provide an option to make masks have the same filename as the input. Probably in 1 hour, stay tuned.

**Reply by sahildeshpande** [reply]
Awesome cheers. Maybe a new feature could be to automate the process instead of manual.

**Reply by continue-revolution** [reply]
I have to think about how to do it because there might be 3 masks and when you want to have the same filename then the masks generated earlier for a particular image might be overwritten by the masks generated later. Probably you have to wait longer.

I would highly recommend generating masks then clicking "generate" because I tend to not create duplicate features, and if you want to do batch mask+inpainting at the same time, you have to enable some sort of script at the bottom of the page, which might not be very clean. Feel free to provide any suggestion on how I can implement this feature.

**Reply by sahildeshpande** [reply]
Yeah true. Fair enough. Best to keep them separate for this extension.
